lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -642,18 +645,65 @@ int ReadDxcOpts(const OptTable *optionTable, unsigned flagsToInclude,
     return 1;
   }
 
+  llvm::StringRef valVersionStr = Args.getLastArgValue(OPT_validator_version);
+  if (!valVersionStr.empty()) {
+    // Parse "major,minor" version string
+    auto verPair = valVersionStr.split(",");
+    llvm::APInt major, minor;
+    if (verPair.first.getAsInteger(0, major) || verPair.second.getAsInteger(0, minor)) {
+      errors << "Format of validator version is \"<major>,<minor>\" (ex: \"1,4\").";
+      return 1;
+    }
+    uint16_t major64 = major.getLimitedValue();
```

Comment:
major64, but type is uint16_t?

Reply:
Typo fixed.  Also, while I was at it, I completed the option by adding support to the linker, with a test.  I also switched the option to use period instead of comma to separate major from minor for consistency.